rustic-rustfmt.el

```diff
@@ -188,6 +196,38 @@ and it's `cdr' is a list of arguments."
         (kill-buffer proc-buffer)
         (message "Workspace formatted with cargo-fmt.")))))
 
+;;;###autoload
+(defun rustic-format-region (begin end)
+  "Format the current active region using rustfmt.
+
+This operation requires a nightly version of rustfmt.
+"
+  (interactive "r")
+  (unless (or (eq major-mode 'rustic-mode)
+              (eq major-mode 'rustic-macro-expansion-mode))
+    (error "Not a rustic-mode buffer."))
+  (let* ((buf (current-buffer))
+         (file (buffer-file-name buf))
+         (r-begin)
+         (r-end))
+    (cond ((region-active-p)
+           (setq r-begin begin)
+           (setq r-end end))
+          (t
+           (setq r-begin (region-beginning))
+           (setq r-end (region-end))))
+    (let ((start (+ 1 (count-lines 1 r-begin)))
+          (len (- (count-lines r-begin r-end) 1)))
+      (rustic-compilation-process-live t)
+      (rustic-format-start-process
+       'rustic-format-file-sentinel
+       :buffer buf
+       :command
+       (append (list rustic-cargo-bin "+nightly" "fmt" "--")
```

Comment:
So this feature is only available on nightly or why do you pin the toolchain ? I guess nowadays lots of people are using stable instead, so we should cover the case when there's no nightly available.

Reply:
Yep, it only works with rustfmt nightly. It looks like they are close to stabilizing (https://github.com/rust-lang/rustfmt/issues/3397)! I'll try to handle this more gracefully